tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,495 @@
+            }
+            addl_hints: PartialRecordsHints = {}
+            if self.engine.name == 'redshift' and not self.has_scratch_s3_bucket():
+                uses_pandas = True
+            elif self.engine.name == 'mysql':
+                # mysql has no exporter defined, so everything is via Pandas
+                uses_pandas = True
+            elif self.engine.name == 'postgresql':
+                if timeonlyformat in ['HH:MI:SS', 'HH24:MI:SS']:
+                    # ensure we keep other areas of ISO style
+                    addl_hints = {
+                        'dateformat': 'YYYY-MM-DD',
+                        'datetimeformat': 'YYYY-MM-DD HH:MI:SS',
+                        'datetimeformattz': 'YYYY-MM-DD HH:MI:SSOF',
+                    }
+                else:
+                    uses_pandas = True
+            elif self.engine.name == 'vertica':
+                # Make sure our '\n' strings below are valid when comparing output
+                addl_hints['record-terminator'] = '\n'
+                if timeonlyformat not in ['HH:MI:SS', 'HH24:MI:SS']:
+                    uses_pandas = True
+            elif self.engine.name == 'bigquery':
+                # All current export is via Avro
+                uses_pandas = True
+            if uses_pandas:
+                # we're going to need to be sure to use hints that work in Pandas
+                addl_hints.update(pandas_compatible_addl_hints)
+                if 'AM' in timeonlyformat:
+                    # TODO: Add a GitHub issue for this
+                    logger.warning('Cannot export this dateformat using Pandas'
+                                   '--skipping test')
+                    continue
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'timeonlyformat': timeonlyformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            expect_pandas_failure = not self.has_pandas() and uses_pandas
+            try:
+                csv_text = self.unload(column_name='time',
+                                       records_format=records_format)
+                allowed_items = [create_sample(timeonlyformat) + "\n",
+                                 # create_sample(datetimeformattz).replace('-00', '+00') + "\n",
+                                 # # TODO: Should this be necessary?
+                                 # create_sample(datetimeformattz) + ".000000\n",
+                                 # # TODO: Should this be necessary?
+                                 # create_sample(datetimeformattz).replace('-00', '.000000+0000\n'),
+                                 # # TODO: Should this be necessary?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker